rest-api/unicode_csv.py

```diff
@@ -0,0 +1,67 @@
+import csv
+import codecs
+import cStringIO
+
+# Writes CSV files with Unicode encoding. (Standard CSV libraries don't handle Unicode.)
+# Adapted from https://docs.python.org/2/library/csv.html
+
+DELIMITER = ','
+
+class UTF8Recoder:
+  """
+  Iterator that reads an encoded stream and reencodes the input to UTF-8
+  """
+  def __init__(self, f, encoding):
+    self.reader = codecs.getreader(encoding)(f)
+
+  def __iter__(self):
+    return self
+
+  def next(self):
+    return self.reader.next().encode("utf-8")
+
+class UnicodeReader:
+  """
+  A CSV reader which will iterate over lines in the CSV file "f",
+  which is encoded in the given encoding.
+  """
+
+  def __init__(self, f, dialect=csv.excel, encoding="utf-8", **kwds):
+    f = UTF8Recoder(f, encoding)
+    self.reader = csv.reader(f, dialect=dialect, **kwds)
+
+  def next(self):
+    row = self.reader.next()
+    return [unicode(s, "utf-8") for s in row]
```

Comment:
e.g. [these edge cases](https://github.com/jdunck/python-unicodecsv/blob/de3b59bb18032aa75681dc4068b92b4023e9d56d/unicodecsv/py2.py#L122-L1255) for float handling aren't reflected here; I don't know if they matter.

Reply:
Ah, yeah, to be honest I'm not using UnicodeReader at this stage, and we may never use it. So I'll remove this for now, since as you point out, it doesn't handle non-strings properly.